aider/models/model.py

```diff
@@ -0,0 +1,52 @@
+saved_openai = None
+
+
+class Model:
+    name = None
+    edit_format = None
+    max_context_tokens = 0
+    tokenizer = None
+
+    always_available = False
+    use_repo_map = False
+    send_undo_reply = False
+
+    prompt_price = None
+    completion_price = None
+
+    def __init__(self, name, openai=None):
+        global saved_openai
+        if (openai and "openrouter.ai" in openai.api_base):
+            saved_openai = openai
+
+        from .openai import OpenAIModel
+        from .openrouter import OpenRouterModel
+        model = None
+        if saved_openai:
+            model = OpenRouterModel(name, saved_openai)
+        else:
+            model = OpenAIModel(name)
+
+        self.name = model.name
+        self.edit_format = model.edit_format
+        self.max_context_tokens = model.max_context_tokens
+        self.tokenizer = model.tokenizer
+        self.prompt_price = model.prompt_price
+        self.completion_price = model.completion_price
+        self.always_available = model.always_available
+        self.use_repo_map = model.use_repo_map
```

Comment:
Did you consider using a `create(..., **kwargs)` classmethod kind of like the `Coder` classes? I'm not strongly arguing for it, just wondering about your thoughts.

I think the main benefit of that approach is that you return an actual instance of one of the subclasses. So you can override the base class behaviors in the normal way by adding code to the subclass.

Reply:
I remember looking at it briefly and can't recall why I didn't take that approach - agree it's better to return a subclass instance.